doc/release_txt_notes.md

```diff
@@ -0,0 +1,107 @@
+# Writing Notes in a RELEASE.txt File
+
+## Introduction
+
+We have been challenged to write more helpful entries in our RELEASE.txt file
+for new features, improvements, and bug fixes. To help users determine the
+effects of our software changes on their applications, we have developed a
+template that can be used to write release notes. The template is described on
+the rest of this page.
+
+Some of our users face scrutiny from regulatory agencies for their use of
+software they did not develop (aka SOUP). When they use our software, they have
+to be aware of every change we make. For every change we make, they need to
+investigate the effect the change will have on their software. If our change
+affects their application, they may have to have their software retested or
+re-certified.
+
+[SOUP](https://en.wikipedia.org/wiki/Software_of_unknown_pedigree) stands for Software Of Unknown/Uncertain Pedigree/Provenance
+
+
+## When to write a RELEASE.txt note
+
+Generally, a release note must be written for every change that is made to the
+code for which users might see a change in the way the software works. In other
+words, if a user might see a difference in the way the software works, a note
+should be written. By code we mean the text that will be compiled into one of
+the company's software products. The code includes configuration changes and
+changes to tools users might work with to configure and build our software.
+
+A release note does not need to be written for changes to the code that users
+will not see.
```

Comment:
Is this intended to mean "A release note does not need to be written for changes to the code that will have no effect on the way the software works"?

Reply:
I feel like what matters is the user experience. Changing a data structure under the hood would be an effect on the way the software works, but would not be visible to the user, and thus not worthy of a RELEASE.txt note.